### ISSUE TITLE
Segmentation fault when reloading nagios status

### ISSUE THREAD

**Issue by dannywarren** [dannywarren/cnagios#1]
Somewhere after the release of nagios 3.5, cnagios started to segfault when reloading the status.dat file.

Backtrace from the segfault:

```
#0  0x0000000801754e00 in sbrk () from /lib/libc.so.7
#1  0x000000080175042e in sbrk () from /lib/libc.so.7
#2  0x00000008017507af in sbrk () from /lib/libc.so.7
#3  0x000000080175bdf5 in free () from /lib/libc.so.7
#4  0x0000000000404a8a in read_status () at read.c:90
#5  0x00000000004051df in getch_loop () at cmds.c:183
#6  0x0000000000403255 in main (argc=<value optimized out>, argv=<value optimized out>) at cnagios.c:253
```

Relevant section of read.c:

```
for ( i = 0; i < service_list_size; i++ ) {
  for ( j = 0; j < STATUS_LIST_ENTRY_SIZE; j++ ) {
    if ( j == LAST_UPDATE ) { continue; }
    if ( j == LAST_STATE_CHANGE_INT ) { continue; }
    free(service_list[i][j]);
    service_list[i][j] = NULL;
  }
}
```

**Comment by dannywarren** [comment]
There is also a different segfault when exiting:

```
#0  0x0000000801754e00 in sbrk () from /lib/libc.so.7
#1  0x000000080175042e in sbrk () from /lib/libc.so.7
#2  0x00000008017507af in sbrk () from /lib/libc.so.7
#3  0x000000080175bdf5 in free () from /lib/libc.so.7
#4  0x0000000800dea559 in Perl_PerlIO_close () from /usr/local/lib/perl5/5.20/mach/CORE/libperl.so.5.20
#5  0x0000000800dea481 in PerlIO_cleantable () from /usr/local/lib/perl5/5.20/mach/CORE/libperl.so.5.20
#6  0x0000000800dee134 in PerlIO_cleanup () from /usr/local/lib/perl5/5.20/mach/CORE/libperl.so.5.20
#7  0x0000000800ceaf1e in perl_destruct () from /usr/local/lib/perl5/5.20/mach/CORE/libperl.so.5.20
#8  0x000000000040689e in perl_hook_free () at perl.c:50
#9  0x0000000000403261 in main (argc=<value optimized out>, argv=<value optimized out>) at cnagios.c:259
```

**Comment by dannywarren** [comment]
And a different one when running cnagios in debug mode:

```
$ ./cnagios -d
Cnagios version 0.31
polling interval is 90 seconds
pad is 1, header_pad is 2

read_status()...
read_v23_status()...
status file is /var/spool/nagios/status.dat
 reading status.dat...last update is Sat Aug  8 15:37:53 (1439073473)
HOST NUMBER 0...
  host_name is ""
  current_state is
Segmentation fault (core dumped)
```

Backtrace:

```
#0  0x00000008017e8fbb in strlen () from /lib/libc.so.7
#1  0x00000008017de535 in vfprintf_l () from /lib/libc.so.7
#2  0x00000008017dc650 in vfprintf_l () from /lib/libc.so.7
#3  0x00000008017dc4a4 in vfprintf_l () from /lib/libc.so.7
#4  0x00000008017cd3b0 in fprintf () from /lib/libc.so.7
#5  0x000000000040506d in debug (msg=0x1 <Address 0x1 out of bounds>, a=<value optimized out>, b=-7704, c=0, d=-9568, e=<value optimized out>, f=-7688,
    g=6495232, h=-7448, i=27586512, j=<value optimized out>, k=<value optimized out>) at debug.c:17
#6  0x0000000000408e9d in read_v23_status () at readv23.c:203
#7  0x0000000000404aee in read_status () at read.c:103
#8  0x000000000040323c in main (argc=<value optimized out>, argv=<value optimized out>) at cnagios.c:250
```

**Comment by dannywarren** [comment]
All three of these fixed in pull #2
